File: LIGHTOJ/1433. Minimum Arc Distance.cpp
#include<bits/stdc++.h>
#include<math.h>

int main()
{
    int i, t, Ox,Oy, Ax,Ay, Bx,By;
    double d,r, theta;

    scanf("%d", &t);

    for(i=1;i<=t;i++)
    {
        scanf("%d %d %d %d %d %d", &Ox,&Oy, &Ax,&Ay, &Bx,&By);

        d = sqrt(pow(Ax-Bx, 2) + pow(Ay-By,2));
        r = sqrt(pow(Ox-Bx, 2) + pow(Oy-By,2));
        theta = acos(1 - pow(d,2)/(2*pow(r,2)));

        printf("Case %d: %lf\n", i, r*theta);

    }
}

File: UVA/1433. Minimum Arc Distance.py
import math

inp=lambda:map(int,input().split())

t = int(input())

for i in range(1,t+1):
    Ox,Oy, Ax,Ay, Bx,By = inp()

    d = math.sqrt((Ax-Bx)**2 + (Ay-By)**2)
    r = math.sqrt((Ax-Ox)**2 + (Ay-Oy)**2)
    theta = math.acos(1 - d**2/(2*r**2))

    print("Case %d" %i,":",r*theta)